compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -376,43 +376,13 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
     }
 
     fn visit_stmt(&mut self, stmt: &'a ast::Stmt) {
-        if let ast::StmtKind::Semi(expr) = &stmt.kind
-            && let ast::ExprKind::Assign(lhs, _, _) = &expr.kind
-            && let ast::ExprKind::Type(..) = lhs.kind
-            && self.sess.parse_sess.span_diagnostic.err_count() == 0
-            && !self.features.type_ascription
-            && !lhs.span.allows_unstable(sym::type_ascription)
-        {
-            // When we encounter a statement of the form `foo: Ty = val;`, this will emit a type
-            // ascription error, but the likely intention was to write a `let` statement. (#78907).
-            feature_err(
-                &self.sess.parse_sess,
-                sym::type_ascription,
-                lhs.span,
-                "type ascription is experimental",
-            ).span_suggestion_verbose(
-                lhs.span.shrink_to_lo(),
-                "you might have meant to introduce a new binding",
-                "let ",
-                Applicability::MachineApplicable,
-            ).emit();
-        }
         visit::walk_stmt(self, stmt);
     }
 
     fn visit_expr(&mut self, e: &'a ast::Expr) {
         match e.kind {
             ast::ExprKind::Type(..) => {
-                if self.sess.parse_sess.span_diagnostic.err_count() == 0 {
-                    // To avoid noise about type ascription in common syntax errors,
-                    // only emit if it is the *only* error.
-                    gate_feature_post!(
-                        &self,
-                        type_ascription,
-                        e.span,
-                        "type ascription is experimental"
-                    );
-                } else {
+                if self.sess.parse_sess.span_diagnostic.err_count() > 0 {
```

Comment:
Can this be removed now? At least the comment below doesn't make sense without it.